google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1263,18 +1299,92 @@ def query_standard_sql?
         #
-        # @return [Array<String>] An array containing Google Cloud Storage URIs
-        #   and/or inline source code.
+        # @return [Array<String>, nil] An array containing Google Cloud Storage URIs
+        #   and/or inline source code, or `nil` if not a logical view.
         #
         # @!group Attributes
         #
         def query_udfs
+          return nil unless @gapi.view
           udfs_gapi = @gapi.view.user_defined_function_resources
           return [] if udfs_gapi.nil?
           Array(udfs_gapi).map { |udf| udf.inline_code || udf.resource_uri }
         end
 
+        ##
+        # Whether automatic refresh of the materialized view is enabled. When true,
+        # the materialized view is updated when the base table is updated. The default
+        # value is true. See {#materialized_view?}.
+        #
+        # @return [Boolean, nil] `true` when automatic refresh is enabled, `false` otherwise;
+        #   or `nil` if not a materialized view.
+        #
+        # @!group Attributes
+        #
+        def enable_refresh?
+          return nil unless @gapi.materialized_view
+          val = @gapi.materialized_view.enable_refresh
+          return true if val.nil?
+          val
+        end
+
+        ##
+        # Sets whether automatic refresh of the materialized view is enabled. When true,
+        # the materialized view is updated when the base table is updated. See {#materialized_view?}.
+        #
+        # @param [Boolean] new_enable_refresh `true` when automatic refresh is enabled, `false` otherwise.
+        #
+        # @!group Attributes
+        #
+        def enable_refresh= new_enable_refresh
+          @gapi.materialized_view = Google::Apis::BigqueryV2::MaterializedViewDefinition.new(
+            enable_refresh: new_enable_refresh
+          )
+          patch_gapi! :materialized_view
+        end
+
+        ##
+        # The time when the materialized view was last modified.
+        # See {#materialized_view?}.
+        #
+        # @return [Time, nil] The time, or `nil` if not present or not a materialized view.
+        #
+        # @!group Attributes
+        #
+        def last_refresh_time
+          Convert.millis_to_time @gapi.materialized_view&.last_refresh_time
+        end
+
+        ##
+        # The maximum frequency at which the materialized view will be refreshed.
+        # The default value is `1800000` (30 minutes). See {#materialized_view?}.
+        #
+        # @return [Integer, nil] The maximum frequency in milliseconds;
+        #   or `nil` if not a materialized view.
+        #
+        # @!group Attributes
+        #
+        def refresh_interval_ms
+          @gapi.materialized_view&.refresh_interval_ms
+        end
+
+        ##
+        # Sets the maximum frequency at which the materialized view will be refreshed.
+        # See {#materialized_view?}.
+        #
+        # @param [Integer] new_refresh_interval_ms The maximum frequency in milliseconds.
```

Comment:
Is it okay to pass `nil`? Would that reset it back to default? (If so, we should say so and say what the default is.)

Reply:
Sending `nil` in the patch appears to result in the attribute remaining unchanged:

```
#<Google::Apis::BigqueryV2::Table:0x00007fe5e845cad8 @etag="iW9q0sYkndjSweMbW+Ou+g==", @materialized_view=#<Google::Apis::BigqueryV2::MaterializedViewDefinition:0x00007fe5e845cc18 @refresh_interval_ms=nil>>
F

Finished in 6.890545s, 0.1451 runs/s, 5.3697 assertions/s.

  1) Failure:
Google::Cloud::Bigquery::Table::materialized_view::bigquery#test_0001_creates, gets, updates and deletes a materialized view [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigquery/acceptance/bigquery/materialized_view_test.rb:112]:
Expected: 1800000
  Actual: 3600000
```

Reply:
I won't make any changes here, since sending `nil` is pointless.